packages/grid/x-data-grid/src/hooks/features/columnHeaders/useGridColumnHeaders.tsx

```diff
@@ -112,6 +112,35 @@ export const useGridColumnHeaders = (props: UseGridColumnHeadersProps) => {
     apiRef.current.columnHeadersContainerElementRef!.current!.scrollLeft = 0;
   }, [apiRef]);
 
+  const scrollOverColumnHeaders = rootProps.experimentalFeatures?.scrollOverColumnHeaders;
+  React.useEffect(() => {
+    if (!scrollOverColumnHeaders) {
+      // TODO: uncomment before merging
+      // return undefined;
+    }
+    const rootElement = innerRef?.current;
+    if (!rootElement) {
+      return undefined;
+    }
+
+    const callback = (event: WheelEvent) => {
+      const virtualScroller = apiRef.current.virtualScrollerRef?.current;
+      if (!virtualScroller) {
+        return;
+      }
+
+      if (event.deltaX !== 0) {
+        event.preventDefault();
```

Comment:
If the `preventDefault` is to prevent going to the previous page if the user scrolls left (when already on the leftmost position), I'll vote not to add this, as this will negate the default browser behavior/user expectation. 

For example when I try to scroll to the left when already on the leftmost column in the virtual scroller it goes to the previous page, the same is the case somewhere on the page (outside the grid), so my expectation would be the same for the header area too.

![grid-swipe-left](https://user-images.githubusercontent.com/12609561/227173988-95782c8b-f483-4150-bd30-e02e769680b9.gif)

Reply:
The issue is that `deltaX` may not be equal to `0`, even if the element is not scrolling. This is causing `preventDefault` to be called unnecessarily every time. To solve this problem, we can replace `deltaX` with `prevScrollLeft.current`, which holds the previous scroll position. This ensures that `preventDefault` is only called when the element is actually scrolling.

```jsx
if (prevScrollLeft.current !== 0) 
        event.preventDefault();
```
https://user-images.githubusercontent.com/34584266/227527127-0424a09c-dce3-453b-8190-58644ca45d7a.mp4

Reply:
I tried to call `preventDefault` conditionally, and it kind of works, but sometimes calling preventDefault doesn't prevent the default browser action:

https://user-images.githubusercontent.com/13808724/235145377-9bef8a39-b1f3-4f28-ba6c-81b76627967b.mov

It's very unreliable and I see two options here:
1. Roll back to always call `preventDefault`
2. Close this PR and potentially look for a better solution to the problem

I would prefer option 1, because it's better than nothing, but I understand that it's not perfect :)
What do you think?
@MBilalShafi 
@joserodolfofreitas